### ISSUE TITLE
Confidence Intervals occasionally wonky

### ISSUE THREAD

**Issue by staticfloat** [johnmyleswhite/Benchmarks.jl#29]
I've noticed that every now and then, the confidence intervals give a strange result.  I print the confidence intervals of every test [if they're both not null](https://github.com/staticfloat/Perftests.jl/blob/master/src/Perftests.jl#L48-L52), which I thought would be enough to guard myself against strangeness, but apparently it's not; occasionally I get `NaN` CIs when the test time is around 4 seconds (and we're still only allotting 10s for the entire benchmark).  Note that the tests that run for 6 seconds don't seem to have this problem.

Here's [an example log](http://buildbot.e.ip.saba.us:8010/builders/perf_nanosoldier-x64/builds/7/steps/Run%20perf%20tests/logs/stdio) demonstrating what I'm seeing.

**Comment by mbauman** [comment]
Caught one:

``` jl
julia> @benchmark _dense2sparsevec(1:10000,100)
================ Benchmark Results ========================
     Time per evaluation: 321.93 μs [-245188.36 ns, 889.05 μs]
Proportion of time in GC: 1.95% [0.00%, 10.20%]
        Memory allocated: 399.09 kb
   Number of allocations: 17 allocations
       Number of samples: 100
   Number of evaluations: 100
 Time spent benchmarking: 0.05 s
```

The elapsed times distribution is rather skewed:

![screen shot 2015-10-07 at 4 08 37 pm](https://cloud.githubusercontent.com/assets/154641/10349698/b74b60cc-6d0d-11e5-8986-7e2ebc672c1b.png)

I can send you the JLD or csv of the results object, if you'd like to look at it closer.

**Comment by johnmyleswhite** [comment]
A CSV file would be great

**Comment by staticfloat** [comment]
Matt...... how did you plot in your terminal?

**Comment by mbauman** [comment]
TerminalExtensions.jl + iTerm2 beta

**Comment by johnmyleswhite** [comment]
Ah, I finally realized that what's been confusing me about this is that the units of measurement are different for the lower and upper bound -- the CI's are symmetric, they just don't look like they are. The only easy fix for this is to cap the confidence intervals at 0 as a lower bound. This kind of mixture model data with intermittent GC isn't really tractable without some heuristic for identifying outliers and then doing a conditional analysis. We could potentially make some progress with winsorization at something like the 90th percentile, but I'd want to make sure we don't cause a decrease in our ability to measure real changes by doing winsorization.

**Comment by johnmyleswhite** [comment]
Will be closed by #31.
